kotlinx.fuzz.api/src/main/kotlin/kotlinx/fuzz/KFuzzConfig.kt

```diff
@@ -209,3 +254,13 @@ private fun KProperty1<KFuzzConfigImpl, *>.asKFuzzConfigProperty(delegate: KFuzz
 private fun KFuzzConfigImpl.configProperties(): List<KFuzzConfigProperty<*>> =
     KFuzzConfigImpl::class.memberProperties
         .map { it.asKFuzzConfigProperty(this) }
+
+private inline fun <T : KFuzzConfig> wrapConfigErrors(buildConfig: () -> T): T = try {
+    buildConfig()
+} catch (e: Throwable) {
+    if (e is ConfigurationException) {
```

Comment:
We can extract `throw` from `if - else`:
```kotlin
    throw if (e is ConfigurationException) {
        e
    } else {
        ConfigurationException("cannot create config", e)
    }
```

I also prefer to use `when` instead of `if - else`, because I find it more readable (and a bit shorter):
```kotlin
throw when (e) {
    is ConfigurationException -> e
    else -> ConfigurationException("cannot create config", e)
}
```

But it is up to you